avaframe/com1DFA/deriveParameterSet.py

```diff
@@ -197,3 +194,245 @@ def checkRelEntThVariation(cfg, variationDict):
+    itemsL = value.split('$')
+    if 'Percent' in key:
+        if '-' in itemsL[0]:
+            itemsP = itemsL[0].split('-')[1]
+            percentsStart = 1. - float(itemsP) / 100.
+            percentsStop = 1. - float(itemsP) / 100.
+        elif '+' in itemsL[0]:
+            itemsP = itemsL[0].split('+')[1]
+            percentsStart = 1. + float(itemsP) / 100.
+            percentsStop = 1. + float(itemsP) / 100.
+        else:
+            percentsStart = 1. - float(itemsL[0]) / 100.
+            percentsStop = 1. + float(itemsL[0]) / 100.
+        # get number of steps
+        steps = int(itemsL[1])
+        # create array with variation factor
+        itemsArray = np.linspace(percentsStart, percentsStop, steps)
+    # TODO add standard value to range
+    elif 'Range' in key:
+        itemsArray = np.linspace(-1.*itemsL[0], itemsL[0], itemsL[1])
+
+    if fullCfg['GENERAL'].getboolean('addStandardConfig'):
+        if 1 not in itemsArray:
+            itemsArray = np.append(itemsArray, 1.)
+            log.info('Standard configuration value added for %s' % (key))
+
+    return itemsArray
+
+
+def setThicknessValueFromVariation(key, cfg, simType, row):
+    """ set thickness value for thickness parameter for all features if multiple according to
+        desired variation
+
+        Parameters
+        ------------
+        key: str
+            thickness variation info
+
+        Returns
+        --------
+        cfg: dict
+            updated dict with info on thickness
+
+    """
+
+    # only add entries to cfg if appropriate for chosen simType (e.g. entTh if ent or entres run)
+    entCondition = (key == 'entThPercentVariation' and 'ent' in simType)
+    secRelCondition = (key == 'secondaryRelThPercentVariation' and cfg['GENERAL']['secRelArea'] == 'True')
+    relCondition = (key == 'relThPercentVariation')
+
+    # fetch variation factor
+    variationFactor = float(row._asdict()[key])
+
+    # update thickness values accoridng to variation
+    if entCondition or secRelCondition or relCondition:
+        thType = key.split('Percent')[0]
+        thFlag = thType + 'FromShp'
+
+        # add thickness values for all features if thFromShape = True
+        if cfg['GENERAL'][thFlag] == 'True':
+            thId = thType + 'Id'
+            thThickness = thType + 'Thickness'
+            thicknessList = cfg['INPUT'][thThickness].split('|')
+            idList = cfg['INPUT'][thId].split('|')
+            for count, id in enumerate(idList):
+                thNameId = thType + id
+                cfg['GENERAL'][thNameId] = str(float(thicknessList[count]) * variationFactor)
+
+            # set percentVaration parameter to actual variation in percent$steps
+            if variationFactor == 1.:
+                # if variation is 0% set percentVaration = ''
+                variationIni = ''
+            elif variationFactor < 1:
+                variationIni = '-' + str((1. - variationFactor) * 100) + '$1'
```

Comment:
Am I correct that the lowest allowed percentVariation in % is then 1%? Do we need to mention this?

Reply:
the percent variation can be anything between 0 and 1

Reply:
the 1 here corresponds to the variationFactor so 1 means value x1 - no variation or 0% variation - if the factor is 0.5 - 50% percent variation, ..